Catalog ingestion script: Parallel ingestion + multiple files at once

Theodlz opened the pull request:
This PR allows us to specify a directory and not just one file, so we can ingest multiple files into one mongo collection. 
Also, it makes sure that the ingestion can process multiple files at once.

bfhealy reviewed (approved):
Nice, this runs as expected locally for me! I was just going to request that the results be printed after ingestion, but I see that's been added in the second commit.

Theodlz commented:
Thanks for the review @bfhealy. I added some more code to make it clearer (mostly just print new files being processed and failures, not all of the verbose (finished blablabla, connected to db...), and fixed some other issue.
